java-client/src/main/java/org/opensearch/client/transport/aws/AwsSdk2Transport.java

```diff
@@ -257,6 +207,8 @@ public <RequestT, ResponseT, ErrorT> ResponseT performRequest(
             } catch (InterruptedException e) {
                 throw new IOException("HttpRequest was interrupted", e);
             }
+        } else {
```

Comment:
This `else` is probably redundant at this point, right?

Reply:
If you remove it, there will be an error that the method must return a value, and I don't want a catch all `else` where we make an assumption on the type of connection instance.